File: core/commands/registry/irules/ws__enabled.py
# Enriched from F5 iRules reference documentation.
"""WS::enabled -- This command can be used to turn off WebSocket processing."""

from __future__ import annotations

from ....compiler.side_effects import ConnectionSide, SideEffect, SideEffectTarget
from .._base import CommandDef, make_av
from ..models import CommandSpec, FormKind, FormSpec, HoverSnippet, ValidationSpec
from ..namespace_models import EventRequires
from ..signatures import Arity
from ._base import _IRULES_ONLY, register

_SOURCE = "https://clouddocs.f5.com/api/irules/WS__enabled.html"


_av = make_av(_SOURCE)


@register
class WsEnabledCommand(CommandDef):
    name = "WS::enabled"

    @classmethod
    def spec(cls) -> CommandSpec:
        return CommandSpec(
            name="WS::enabled",
            dialects=_IRULES_ONLY,
            hover=HoverSnippet(
                summary="This command can be used to turn off WebSocket processing.",
                synopsis=("WS::enabled ( 'false' )?",),
                snippet=(
                    "WS::enabled\n"
                    "    This can be used to determine whether the Websocket processing is enabled or disabled for a particular connection.\n"
                    "\n"
                    "WS::enabled false\n"
                    "    This can be used to disable the Websocket processing for a particular connection."
                ),
                source=_SOURCE,
                examples=("when HTTP_REQUEST {\n    WS::enabled false\n}"),
                return_value="WS::enabled command returns TRUE if Websocket processing is enabled for a particular connection, FALSE otherwise.",
            ),
            forms=(
                FormSpec(
                    kind=FormKind.DEFAULT,
                    synopsis="WS::enabled ( 'false' )?",
                    arg_values={
                        0: (_av("false", "WS::enabled false", "WS::enabled ( 'false' )?"),)
                    },
                ),
            ),
            validation=ValidationSpec(
                arity=Arity(),
            ),
            event_requires=EventRequires(profiles=frozenset({"HTTP"})),
            side_effect_hints=(
                SideEffect(
                    target=SideEffectTarget.NETWORK_IO,
                    reads=True,
                    connection_side=ConnectionSide.BOTH,
                ),
            ),
        )

File: rust/tcl-registry/src/commands/irules/ws__enabled.rs
// tcl-lsp — a language server and toolchain for Tcl
// Copyright (C) 2026 James Deucker (bitwisecook) <https://github.com/bitwisecook>
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//
// SPDX-License-Identifier: AGPL-3.0-or-later

//! `WS::enabled` iRules command.
use crate::prelude::*;
pub const fn spec() -> CommandSpec {
    CommandSpec {
        name: "WS::enabled",
        dialects: Some(DialectSet::IRULES),
        arity: Arity::at_least(0),
        hover: Some(HoverSnippet {
            summary: "This command can be used to turn off WebSocket processing.",
            synopsis: &["WS::enabled ( 'false' )?"],
            snippet: "WS::enabled\n    This can be used to determine whether the Websocket processing is enabled or disabled for a particular connection.\n\nWS::enabled false\n    This can be used to disable the Websocket processing for a particular connection.",
            source: "https://clouddocs.f5.com/api/irules/WS__enabled.html",
            examples: "when HTTP_REQUEST {\n    WS::enabled false\n}",
            return_value: "WS::enabled command returns TRUE if Websocket processing is enabled for a particular connection, FALSE otherwise.",
        }),
        event_requires: Some(EventRequires {
            client_side: false,
            server_side: false,
            transport: None,
            profiles: &["HTTP"],
            also_in: &[],
            init_only: false,
            flow: false,
            capability: None,
        }),
        forms: &[FormSpec {
            kind: FormKind::Default,
            synopsis: "WS::enabled ( 'false' )?",
            dialects: None,
        }],
        side_effects: &[SideEffect {
            target: SideEffectTarget::NetworkIo,
            reads: true,
            writes: false,
            connection_side: ConnectionSide::Both,
            dialects: None,
        }],
        ..CommandSpec::DEFAULT
    }
}
